ops/framework.py

```diff
@@ -784,16 +784,16 @@ class SomeObject:
         if not isinstance(bound_event, BoundEvent):
             raise RuntimeError(
                 f'Framework.observe requires a BoundEvent as second parameter, got {bound_event}')
-        if not isinstance(observer, types.MethodType):
-            # help users of older versions of the framework
-            if isinstance(observer, charm.CharmBase):
-                raise TypeError(
-                    'observer methods must now be explicitly provided;'
-                    ' please replace observe(self.on.{0}, self)'
-                    ' with e.g. observe(self.on.{0}, self._on_{0})'.format(
-                        bound_event.event_kind))
+        # Help users of older versions of the framework.
+        if isinstance(observer, charm.CharmBase):
+            raise TypeError(
+                'observer methods must now be explicitly provided;'
+                ' please replace observe(self.on.{0}, self)'
+                ' with e.g. observe(self.on.{0}, self._on_{0})'.format(
+                    bound_event.event_kind))
+        if not callable(observer):
             raise RuntimeError(
-                f'Framework.observe requires a method as third parameter, got {observer}')
+                f'Framework.observe requires a callable as third parameter, got {observer}')
```

Comment:
I made these changes while figuring out the lambda case (as noted in #1129, it's non-trivial to make that work, because we need the handler to be attached to an `Object`), and they aren't required to solve the actual problem.

I could therefore revert these changes to make the PR smaller, although I feel like this is more what we want - can you call the observer with the appropriate arguments - so I'm tempted to leave it. Please push back as needed :smile:.

Reply:
Let's get rid of the "older versions" check (but ensure it would fail lower down), and revert `if not callable` to `if not MethodType`. Let's also make it `TypeError` rather than runtime error.

And let's avoid use of confusing "third" (it includes self) -- name the parameter instead.